File: src/Holefeeder.Ui.Shared/Models/Category.cs
namespace Holefeeder.Ui.Shared.Models;

public class Category
{
    public Guid Id { get; init; }
    public required string Name { get; init; }
    public string Color { get; init; } = string.Empty;
    public decimal BudgetAmount { get; init; }
    public bool Favorite { get; init; }
}

File: src/Holefeeder.Ui/lib/models/category.dart
import 'package:decimal/decimal.dart';

class Category {
  final String id;
  final String name;
  final String color;
  final Decimal budgetAmount;
  final bool favorite;

  const Category({
    required this.id,
    required this.name,
    required this.color,
    required this.budgetAmount,
    required this.favorite,
  });
}
